File: ios/SecretaryVaultAccess/Sources/SecretaryVaultAccessTesting/FakeSyncMonitorHook.swift
import Foundation
import SecretaryVaultAccess

/// Spy `SyncMonitorHook` for VM tests.
@MainActor
public final class FakeSyncMonitorHook: SyncMonitorHook {
    public private(set) var muteCalls = 0
    public private(set) var acknowledgeCalls = 0
    public init() {}
    public func muteSelfWrite() { muteCalls += 1 }
    public func acknowledge() { acknowledgeCalls += 1 }
}

File: android/vault-access/src/test/kotlin/org/secretary/sync/FakeSyncMonitorHook.kt
package org.secretary.sync

/** Host-test [SyncMonitorHook] that counts forwarded calls (spy). */
class FakeSyncMonitorHook : SyncMonitorHook {
    var muteCount: Int = 0
        private set
    var acknowledgeCount: Int = 0
        private set

    override fun muteSelfWrite() {
        muteCount++
    }

    override fun acknowledge() {
        acknowledgeCount++
    }
}
